lean4-infoview/src/infoview/rpcSessions.tsx

```diff
@@ -219,7 +220,7 @@ export class RpcSessions implements Disposable {
         this.#connected.delete(uri)
         this.#connecting.get(uri)?.then(sesh => {
             sesh.dispose()
-            this.#connected.delete(uri)
+            this.#connecting.delete(uri)
```

Comment:
I think this line, on the other hand, was correct. Note that when a new promise is inserted into `#connecting`, it already includes code that will remove it from `#connecting` when the promise resolves, but will add it to `#connected`. Hence the `#connected.delete` here. I spent quite a while thinking about this design and couldn't find a way to simplify it, unfortunately :/

Reply:
Ok, I reverted it, but I'm still worried that sessionAt actually looks at what is in the this.#connecting, or are you confident that the connectAt newSesh.then() handler will always fire and cleanup the this.#connecting list.  What happens in the case that line 143 `this.#ec.api.sendClientRequest` fails?  Who cleans up the this.#connecting list in that case?

Reply:
I am not, and the request can in fact fail if the file is not open. I started working on a fix for this, which I think I can push directly to this branch.

Reply:
ok, if you have a fix that would be great.  This branch is in my own fork because I don't think I have access to make branches directly on this repo...

Reply:
I pushed something that should make the session manager more reliable in normal usage. On server restart we will still see the `keepAlive`s because we're not yet restarting the infoview.

Reply:
Thanks I like your improvements and tested them they seem to work fine.  I'm not sure about line 197 `this.connectAt(uri);` inside the call catch block because there are times when switching lean server versions where I see this catch block get hit 4 or 5 times in rapid succession.  Perhaps it should wait for the next call to run connectAt.  Also if I just let the editor sit there for a few minutes while I'm doing nothing, I see a series of RpcNeedsReconnect calls.  If it did not automatically reconnect it wouldn't do this and the "idle" would be more "idle":
![image](https://user-images.githubusercontent.com/18707114/140260338-31951074-0c06-47a5-851e-36f2f3fe2522.png)

Reply:
> there are times when switching lean server versions where I see this catch block get hit 4 or 5 times in rapid succession. Perhaps it should wait for the next call to run `connectAt`

Note that the first call to `connectAt` will insert the `uri` into `#connecting`, so on further `RpcNeedsReconnect` exceptions we'll just return `undefined` but will not try to connect again.

Reply:
> Also if I just let the editor sit there for a few minutes while I'm doing nothing, I see a series of `RpcNeedsReconnect` calls. If it did not automatically reconnect it wouldn't do this and the "idle" would be more "idle":

The client should always try to reconnect if it gets the `RpcNeedsReconnect` error from the server, but the server should not send this unless it has crashed or restarted. So seeing this when idling is a bug, but one that we should fix by ensuring the session stays alive rather than removing the reconnection code.

Reply:
Ok, and I'm seeing that this RpcNeedsReconnect  does not keep happening in your #66